### ISSUE TITLE
curl samples should more precisely describe what to verify

### ISSUE THREAD

**Issue by jmichler** [Entwicklung-AvaL-Standard/oauth2-samples#11]
Hi,

I'd suggest to add to the curl samples, that it is task of the person verifying the token to do the following things:
- verify the "aud" of the response (curl/2_token_introspection_response_valid.txt) is myself
- verify the "client_id" is the partner I suppose is calling me
- verify that the "active"  flag is true

@curban111 is this correct? If so wouldn't it make sense to include this in curl/README.md ?

**Comment by jmichler** [comment]
And if so: How can I get that certificate? Currently my Engine fails to validate due to:
The certificate for the alias 2kgGZS9TJcSmSLjM-6HF6SfjiWlAEJgzsoGDbeBjhOU not found in the KSS Keystore kss://owsm/keystore. The following aliases are found in the Keystore...
Where do I get the certificates for test- and PROD from? And are they "stable"?

**Comment by curban111** [comment]
Yes, we should add more information and descriptions about token verification. Especially because this check cannot be done directly with curl.

In order to answer the question about the trustworthiness (certificate check) of the server, I have to contact the server operator.

**Comment by curban111** [comment]
Last question first. You can retrieve a certificate for PROD and test by using the follwing URLs:

- **PROD:** https://auth.avalstandard.de/auth/realms/avalstandard/protocol/openid-connect/certs
- **test:** https://auth-test.avalstandard.de/auth/realms/avalstandard/protocol/openid-connect/certs

**curl example for PROD**
`$ curl https://auth.avalstandard.de/auth/realms/avalstandard/protocol/openid-connect/certs > auth.avalstandard.de.realm-cert.json`

The certificate is located in the key **x5c** within the retrieved Json file. The URL can also be determined via the "well-known" configuration of the central OIDC OAuth2 server:

- **jwks_uri:** https://auth.avalstandard.de/auth/realms/avalstandard/.well-known/openid-configuration

There is no key rotation at the moment. Given that tokens presently expire after only 5 minutes without the provision of refresh tokens, any future rotation will happen rapidly. However, this shouldn't pose an issue, as you can access the keys online at any time through the provided URL. So there is actually no need to store the tokens locally.

@jmichler Was this information useful to you?
